crates/core/src/factories/core_fetch.rs

```diff
@@ -418,31 +387,33 @@ impl CoreFetch {
                 }
             }
 
-            // Re-insert the fetch request into the queue after a delay.
-            let outgoing_request_tx = outgoing_request_tx.clone();
-
-            tokio::task::spawn({
-                let state = state.clone();
-                async move {
-                    tokio::time::sleep(Duration::from_millis(
-                        re_insert_outgoing_request_delay as u64,
-                    ))
-                    .await;
-                    if let Err(err) = outgoing_request_tx
-                        .try_send((op_id.clone(), peer_url.clone()))
-                    {
-                        tracing::warn!(
+            if !peer_url_unresponsive {
```

Comment:
This is new too. Before requests were always re-inserted into the request queue. Now they're only re-inserted if the peer is not unresponsive. My thinking is that if a URL is unresponsive, there's no point to keep the requests in the queue, and if the URL becomes available again, gossip will happen again and insert missing ops in the request queue again. If the URL is renewed for the peer, it's good that the requests are no longer in the queue.